### DISCUSSION TITLE
What do you suggest in substitution of Processing's selectInput(), selectFolder(), & etc?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#96] [Ideas]
I'm playing with this:
```python
import tkinter
...
file = tkinter.filedialog.askopenfilename()
folder = tkinter.filedialog.askdirectory()
```
![image](https://user-images.githubusercontent.com/3694604/179121963-0d8a63d6-868f-4f99-8351-a7a50429aeca.png)

I don't mind that they don't look very "native" to my OS (Manjaro XFCE), but they are a tad clumsy, old-fashioned...
Any other suggestions?

**Comment by hx2A** [comment]
I have no idea! Ironically, I know very little about Python GUI programming. I've never used tkinter or any other GUI framework.

One idea is for py5 to call Processing's selectInput() or selectFolder(), etc. At least it would be familiar to Processing folks.

There are probably other approaches. For this I'd like to be patient and wait for more ideas to emerge before taking action.

**Reply by villares** [reply]
No need to hurry, I think this is one of the hard ones! There are many complexities... the Processing approach does work, using callbacks and creating a separate thread, but, in my experience in Java and Python modes, it magically silences exceptions (in the callback function, if I remember correctly) in order to avoid exposing IOErrors, I suppose, but it can be very weird, because it silences most other errors! I find it most unpythonic (it has caused me debugging nightmares...). I have also noticed the Processing.py/Python mode dialogs are less nice than the Java mode ones: https://github.com/jdf/Processing.py-Bugs/issues/303

**Reply by hx2A** [reply]
This morning I thought of an approach that might work. The callbacks in Processing's methods are what make this tricky and are why I left them out of py5's API. But a few weeks ago I added a feature that could be used to get this to work. I will create an issue to track this.  See #98
